tests/unit_tests/weightwindows/test.py

```diff
@@ -0,0 +1,169 @@
+
+
+def test_weightwindows(model):
+
+    ww_files = ('ww_n.txt', 'ww_p.txt')
+    cwd = Path(__file__).parent.absolute()
+    filepaths = [cwd / Path(f) for f in ww_files]
+
+    with cdtemp(filepaths):
+        # run once with variance reduction off
+        model.settings.weight_windows_on = False
+        analog_sp = model.run()
+        os.rename(analog_sp, 'statepoint.analog.h5')
+
+        # weight windows
+
+        # load pre-generated weight windows
+        # (created using the same tally as above)
+        ww_n_lower_bnds = np.loadtxt('ww_n.txt')
+        ww_p_lower_bnds = np.loadtxt('ww_p.txt')
+
+        # create a mesh matching the one used
+        # to generate the weight windows
+        ww_mesh = openmc.RegularMesh()
+        ww_mesh.lower_left = (-240, -240, -240)
+        ww_mesh.upper_right = (240, 240, 240)
+        ww_mesh.dimension = (5, 6, 7)
+
+        # energy bounds matching those of the
+        # generated weight windows
+        e_bnds = [0.0, 0.5, 2E7]
+
+        ww_n = openmc.WeightWindows(ww_mesh,
+                                    ww_n_lower_bnds,
+                                    None,
+                                    10.0,
+                                    e_bnds)
+
+        ww_p = openmc.WeightWindows(ww_mesh,
+                                    ww_p_lower_bnds,
+                                    None,
+                                    10.0,
+                                    e_bnds)
+
+        model.settings.weight_windows = [ww_n, ww_p]
+
+
+        # run again with variance reduction on
+        model.settings.weight_windows_on = True
+        ww_sp = model.run()
+        os.rename(ww_sp, 'statepoint.ww.h5')
+
+        # load both statepoints and examine results
+        asp = openmc.StatePoint('statepoint.analog.h5')
+        wsp = openmc.StatePoint('statepoint.ww.h5')
+
+        analog_tally = asp.tallies[1]
+        ww_tally = wsp.tallies[1]
+
+        def compare_results(particle, analog_tally, ww_tally):
+
+            # get values from each of the tallies
+            an_mean = analog_tally.get_values(filters=[openmc.ParticleFilter],
+                                              filter_bins=[(particle,)])
+            ww_mean = ww_tally.get_values(filters=[openmc.ParticleFilter],
+                                          filter_bins=[(particle,)])
+
+            # expect that more bins were scored with weight windows than
+            # the analog run
+            assert np.count_nonzero(an_mean) < np.count_nonzero(ww_mean)
+
+            an_rel_err = analog_tally.get_values(filters=[openmc.ParticleFilter],
+                                                filter_bins=[(particle,)],
+                                                value='rel_err')
+            ww_rel_err = ww_tally.get_values(filters=[openmc.ParticleFilter],
+                                                filter_bins=[(particle,)],
+                                                value='rel_err')
```

Comment:
Hanging indents here should be fixed

Reply:
Would anyone be interested in added an auto PEP8 to the github actions? It could automatically fix things like this. Example one [here](https://github.com/fusion-energy/paramak/blob/main/.github/workflows/black.yml). The black commands can be customized if a longer line length is preferred and the automatic commit has a ```[skip ci]``` flag so it doesn't rerun the tests. Anyway let me know if this would be of interest and I shall made a new PR.